### PR TITLE
CID-18610 Rebuild with next.js and React

### PR DESCRIPTION
**Pull request by lps88**

### CODE REVIEW

File: src/app/_components/article.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            <div className="nhsuk-grid-column-one-third sticky-nav">
+              <div className="article-section-nav-wrapper">
+                <div className="article-section-nav">
+                  <h2 className="article-section-nav__title">{post.title}</h2>
+                  <ArticleNavList
+                    headers={headings
+                      .filter(({ depth }) => depth <= level)
+                      .filter(
+                        ({ className }) =>
+                          !className || className.indexOf('nhsuk-u-visually-hidden') === -1
```
Comment (ri-ch) [inline src/app/_components/article.tsx:28]: For simplicity?

```suggestion
                          !className || !className.includes('nhsuk-u-visually-hidden')
```

File: src/app/_components/footer.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+export default function Footer() {
```
Comment (ri-ch) [inline src/app/_components/footer.tsx:1]: Missing `'use client';`?

File: src/app/_components/nav-list.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  if (startIndex === endIndex) return startIndex;
+  else if (startIndex === endIndex - 1) {
+    if (
+      Math.abs(sectionPositionArray[startIndex].offsetTop - currentPosition) <
+      Math.abs(sectionPositionArray[endIndex].offsetTop - currentPosition)
+    )
+      return startIndex;
+    else return endIndex;
+  } else {
+    var nextNearest = ~~((startIndex + endIndex) / 2);
```
Comment (ri-ch) [inline src/app/_components/nav-list.tsx:31]: Not sure what the linter makes of this, but as far as I can see these could be `const`.

File: src/app/_components/footer.tsx (hunk 2 above)
Comment (lps88) [reply src/app/_components/footer.tsx:1]: This component is not used client side so doesn't need that, it's gonna be statically rendered only once at build time. I was wondering if there was a better naming convention for server components vs client components (especially as we have one of each for the header) but open to suggestions

File: src/app/_components/nav-list.tsx (hunk 3 above)
Comment (lps88) [reply src/app/_components/nav-list.tsx]: yep good shout

File: src/app/_components/article.tsx (hunk 1 above)
Comment (lps88) [reply src/app/_components/article.tsx]: yep that reads better

**Review by ri-ch — APPROVED** [review]

**Review by NumaanGulfrazNHSD — APPROVED** [review]

**Pull request merged**